scripts/util/yuidoc.js

```diff
@@ -0,0 +1,53 @@
+var yuidoc = require('yuidocjs');
+var fs = require('fs');
+var path = require('path');
+var util = require('util');
+
+/** prepare options in format required by YUIDOC
+  * It is based on logic in yuidoc.Project.init
+  */
+var project = yuidoc.Files.getJSON('./yuidoc.json');
+var options = project.options;
+  delete project.options;
+  options.paths = [];
+  options.project = project;
+  options.nocode = true;
+  options.quiet = true;
+  options.themedir = path.dirname(require.resolve(path.join('yuidoc-lucid-theme', 'theme.json')));
+  options.helpers = [path.join(options.themedir, 'helpers', 'helpers.js')];
+
+module.exports = {
+  generate: function (_options) {
+    options = util._extend(options, _options);
+    var starttime = (new Date()).getTime();
+    var promise = Promise.resolve();
```

Comment:
If these can be done in parallel, you can do `Promise.all` instead of chaining the promises in this way.